x-pack/plugins/security_solution/server/lib/detection_engine/routes/utils.ts

```diff
@@ -289,3 +293,47 @@ export const convertToSnakeCase = <T extends Record<string, unknown>>(
     return { ...acc, [newKey]: obj[item] };
   }, {});
 };
+
+export const mergeStatuses = (
+  id: string,
+  failures: Array<SavedObjectsFindResult<IRuleStatusAttributes>>,
+  acc: RuleStatusResponse
+) => {
```

Comment:
Nit: explicit return type here. This might also remove the need for the `as RuleStatusResponse` below

Reply:
edit: still need the `as` below, but I think the issue that it's resolving is correct (`convertToSnakeCase` is too fuzzy).